### DISCUSSION TITLE
No clock message for alf

### DISCUSSION THREAD

**Discussion by quasiquote1** [sparcians/map#276] [Q&A] [answered]
Hi,

I am creating my own alf file for Helios, and I noticed that I get a "#NO! clock" message where the cycle count should be:
![Screen Shot 2021-10-05 at 08 17 41 AM](https://user-images.githubusercontent.com/85322368/136052530-5480c4c9-7713-453d-b26c-f0d0a1a1d245.png)

Is there a way I can fix this?

Thanks!

**Comment by klingaard** [comment]
Wondering if you can provide a little detail.  Can you share part of the ALF?

**Comment by quasiquote1** [comment]
I have attached my ALF and .cmd files.  

In my simulation I have connected two Sparta Units together (each on their own ResourceTreeNode), and those two Units are connected to the root node.  

I should definitely have a clock object since I see my simulation complete successfully and it also prints out stats for root clock cycles.

[rob.alf.txt](https://github.com/sparcians/map/files/7288178/rob.alf.txt)

[rob_column.cmd.txt](https://github.com/sparcians/map/files/7288180/rob_column.cmd.txt)
[rob.cmd.txt](https://github.com/sparcians/map/files/7288181/rob.cmd.txt)

**Reply by klingaard** [reply]
Thanks, I'll try to find some time today to take a peek at it.

**Comment by klingaard** [comment, marked as answer]
Ok, I see how this works.  @TopChangwatchaiSiFive would understand this best, but your ROB name is different from what's hard-coded in the perl script:
https://github.com/sparcians/map/blob/95d712628b6d733afaa288f32077ed83f6964dbb/helios/pipeViewer/scripts/gen_alf/gen_alf#L691

Specifically, I think you need to change `LOC.rob.ReorderBuffer.ReorderBuffer0` in `gen_alf` to `ROBBuffer_age_ordered.ROBBuffer_age_ordered0` (or something to that effect).   

I'm not 100% sure why you need to use the ROB entry for the cycle count, but that seems to be the problem.  Maybe Top would be able to answer that...

**Comment by quasiquote1** [comment]
Thanks for looking into this.  This was more of an annoyance more than anything - it wasn't affecting functionality since the bottom most cycle controls were still working.
